### PR TITLE
fix: influxdb float serialization error

### PR DESCRIPTION
**Pull request by JimMoen**
Fixes <issue-or-jira-number>

### <samp>🤖 Generated by Copilot at 4a7ba95</samp>

Fix a bug in `emqx_bridge_influxdb_connector` that caused serialization errors when writing integers as floats to the InfluxDB bridge. Add a changelog file `fix-11223.en.md` to document the bug fix.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_bridge_influxdb/src/emqx_bridge_influxdb_connector.erl [hunk 1]
```diff
@@ -638,8 +638,10 @@ value_type([UInt, <<"u">>]) when
     is_integer(UInt)
 ->
     {uint, UInt};
-value_type([Float]) when is_float(Float) ->
-    Float;
+%% write `1`, `1.0`, `-1.0` all as float
+%% see also: https://docs.influxdata.com/influxdb/v2.7/reference/syntax/line-protocol/#float
+value_type([Number]) when is_number(Number) ->
+    Number;
```
Comment (savonarola) [inline apps/emqx_bridge_influxdb/src/emqx_bridge_influxdb_connector.erl:644]: shouldn't we convert to float then?

Comment (JimMoen) [reply apps/emqx_bridge_influxdb/src/emqx_bridge_influxdb_connector.erl:644]: In fact, if you write an integer without a decimal point as a field value, influxdb will think it is a float value, unless `i` or `u` is added after it when serializing the line protocol to let influxdb understand that this is an int or uint value.
We should not change its original literal value while writing.
Of course, the current indirect method is to multiply the value that needs to be written for the float type by 1.0 in the rule engine.

See this [line-protocol-float-example](https://docs.influxdata.com/influxdb/v2.7/reference/syntax/line-protocol/#float-field-value-examples)

Comment (HJianBo) [reply apps/emqx_bridge_influxdb/src/emqx_bridge_influxdb_connector.erl:644]: I see.  SO, according to the InfluxDB protocol, the *Integer* literal value will actually be treated as a `Float` value.

**Review by HJianBo — APPROVED** [review]
LGTM

**Review by savonarola — APPROVED** [review]

**Pull request merged**